### ISSUE TITLE
Light spot in trainning.

### ISSUE THREAD

**Issue by 594cp** [NVIDIA/pix2pixHD#46]
Hi,I  have a problem in my trainning.It works well when I train 256x256 images using train_A to train_B,but when I train  512x1024 images using Label to image without instance map,I have a light spot in every generated images,what's wrong?

![-1](https://user-images.githubusercontent.com/33000834/43053850-6bd6fca0-8e61-11e8-999f-83d5566acf72.jpg)

the red rectangle shows the light spot position.

![-2](https://user-images.githubusercontent.com/33000834/43053917-bf00a2aa-8e61-11e8-9a1b-db3df030562d.jpg)
Anyone  has the same problems?

**Comment by tcwang0509** [comment]
As addressed by @Quasimondo, this problem can be alleviated if you replace all zero paddings in the global generator with reflection paddings.

**Comment by Quasimondo** [comment]
I should add that changing the padding does not solve the artifact problem entirely, but it seems that there are less of them. Overall this is indeed a problem that bothers me quite a bit since once a model has "caught" these artifacts (which I believe originate from the residual layers) they seems to hamper the learning process considerably since they bias the normalization. You might have noticed that frames that show these artifacts are usually quite flat  and without contrast in the other areas. Also further training does usually just increase the amount of artifacts.

One experimental approach to "heal" a model that has caught them is to increase the learning rate a lot, e.g. 0.0008 instead of 0.0002. But then it is important to observe the training results and stopping the process once the artifacts seem to have disappeared. Of course this is a dangerous path and you should keep a backup of your model since if the gradients explode you model will be ruined.

**Comment by ksteinfe** [comment]
Same issue as described above.

@tcwang0509, would you mind detailing how to go about replacing the zero paddings with reflection paddings?

I see that the default `padding_type` argument on of the `__init__` method of the `GlobalGenerator` class is already "reflect". Does that default setting a result of a change that has been made since the start of this thread? 

```
class GlobalGenerator(nn.Module):
    def __init__(self, input_nc, output_nc, ngf=64, n_downsampling=3, n_blocks=9, norm_layer=nn.BatchNorm2d, 
                 padding_type='reflect'):
```

Or are changes needed on the following lines, which call `nn.Conv2d` with `padding=0`?

line 192
`model = [nn.ReflectionPad2d(3), nn.Conv2d(input_nc, ngf, kernel_size=7, padding=0), norm_layer(ngf), activation]`
line 209
`model += [nn.ReflectionPad2d(3), nn.Conv2d(ngf, output_nc, kernel_size=7, padding=0), nn.Tanh()]`

Thank you for your help!

**Comment by Quasimondo** [comment]
What I did was adding an nn.ReflectionPadding() module before the
convolution and set the nn.Convolution2d padding to 0

Best,
Mario

**Comment by ksteinfe** [comment]
Like this?
starting on line 193...
```
        ### downsample
        for i in range(n_downsampling):
            mult = 2**i
            model += [nn.ReflectionPad2d(3), nn.Conv2d(ngf * mult, ngf * mult * 2, kernel_size=3, stride=2, padding=0),
                      norm_layer(ngf * mult * 2), activation]
```

**Comment by Quasimondo** [comment]
Yes, that looks right to me.

**Comment by aviel08** [comment]
Hey @ksteinfe are you sure that code is right? it gives me an error when I try it. I changed the reflection type to 1.
I think the order is:
1=Reflect
2=Replicate
3=zero
> 
> ```
>    model += [nn.ReflectionPad2d(1), nn.Conv2d(ngf * mult, ngf * mult * 2, kernel_size=3, 
> ```

**Comment by ytsheng** [comment]
@aviel08 the input is just the padding size: https://pytorch.org/docs/master/generated/torch.nn.ReflectionPad2d.html

**Comment by chadify007** [comment]
> One experimental approach to "heal" a model that has caught them is to increase the learning rate a lot, e.g. 0.0008 instead of 0.0002.

What about starting with a lower learning rate earlier on instead?
